test/unit/mdc-textfield/foundation.test.js

```diff
@@ -204,19 +204,15 @@ test('#setValid updates classes', () => {
 
 test('#setRequired updates CSS classes', () => {
   // Native validity checking does not apply in unittests, so manually mark as valid or invalid.
-  const {foundation, mockAdapter, nativeInput, helperText} =
+  const {foundation, mockAdapter, nativeInput} =
     setupValueTest('', /* isValid */ false);
 
   foundation.setRequired(true);
   assert.isOk(foundation.isRequired());
-  td.verify(mockAdapter.addClass(cssClasses.INVALID));
```

Comment:
If we don't have a negative test for this (testing from required=true to required=false) we should probably add one (see my comment above).